### PR TITLE
developing with moveit

### PR DESCRIPTION
**Pull request by mlautman**
This has been sitting for a while. It should either be merged or abandoned.

### CODE REVIEW

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
+Developing with MoveIt!
+========================
+Below is an collection of information useful for developing with MoveIt!
+
+Virtual Machines
+-----------------------------------------------------
+MoveIt! is available on Linux. We recommend developing with MoveIt on a native install of Ubuntu 16.04. If you are unable to run Linux natively on your machine, the next best thing would be to set up a virtual machine using VMware. There are known issues with VirtualBox.
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: This does not seem relevant to "developing" and should be removed... this paragraph belongs on the website install page

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+-----------------------------------------------------
+As you add and remove packages from your workspace you will need to clean your workspace and re-run the command to install missing dependancies.
+
+If, for instance, you wanted to install ``moveit_visual_tools`` from source so that you could make changes and add functionality
+::
+
+   cd ~/ws_moveit/src
+   git clone https://github.com/davetcoleman/moveit_visual_tools.git
+
+You would then need to clean your workspace to remove references to the system wide installation of ``moveit_visual_tools``
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: i don't think this is true...

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Tests
+-----
+How to test changes to MoveIt! on various robots, including unit and integration tests.
+
+**Note:** *This is a stub tutorial, to be expanded upon in the future*
+
+Integration Test
+^^^^^^^^^^^^^^^^
+
+A Python-based integration test is available for testing higher-level move_group functionality in MoveIt! - to run ::
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: Explain that there should be many integration tests - is this really the only one?

Then say "A Python-based integration test *example* is available..."

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+From the package `panda_moveit_config <https://github.com/PickNikRobotics/panda_moveit_config>`_ ::
+
+  roslaunch panda_moveit_config demo.launch
+
+**Fanuc M-10iA**
+
+From the package `moveit_resources <https://github.com/ros-planning/moveit_resources>`_ ::
+
+  roslaunch moveit_resources demo.launch
+
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: The PR2 is still in many of our tests so should be mentioned

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Unit Tests
+^^^^^^^^^^
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: i like this section but can you add a link to this page from http://moveit.ros.org/documentation/contributing/ that says "see here" for how to run tests.

this way its not duplicate

File: doc/developing_with_moveit/developing_with_moveit_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+To run unit tests locally on the entire MoveIt! catkin workspace using catkin-tools::
+
+  catkin run_tests -iv
+
+To run a test for just 1 package::
+
+  catkin run_tests --no-deps --this -iv
+
+Kinematic Tests
```
Comment (davetcoleman) [inline doc/developing_with_moveit/developing_with_moveit_tutorial.rst]: can we add a section above this one that moved the section "Using Docker to Review Pull Requests" from [this page](http://moveit.ros.org/documentation/contributing/) to here?

**Comment by davetcoleman** [comment]
@mlautman why was this closed?

**Comment by mlautman** [comment]
I didn't think that my changes here added much to the quality of the tutorials. Is there anything in particular here that you would want to keep? 

I can open a new PR with your requested changes ("using docker to review pull requests", the PR2 stuff and the 'see here to run tests' change)
